check-examples.sh

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+./extract-code-blocks.sh
+
+for f in examples/v*/*/*.go ; do
+    df=$(dirname $f)/$(basename $f .go)
+    mkdir -p $df
+    cp $f $df/main.go
```

Comment:
Do you want to do `mv` here? It's a bit confusing to have both. Or is it for docs purpose?

Reply:
I want to keep both for some time. Ideally, I would delete all `main.go` files after the check, but it helps with debugging locally.